src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -1463,6 +1544,10 @@ impl Decimal {
         }
         res
     }
+
+    pub fn is_zero(&self) -> bool {
+        !self.word_buf.iter().any(|&x| x != 0)
```

Comment:
That's not true. When some decimal is reset to zero, only first byte is reset to 0.